### ISSUE TITLE
Passable does not block vanilla client from falling through blocks

### ISSUE THREAD

**Issue by zedwick** [NoCheatPlus/Issues#143]
What is the problem? What do you expect to see instead?
This is specifically running the HeavySpleef plugin; when breaking blocks whilst "in-game" (pre-game lobby included, just generally tracked by the plugin) attempts to break blocks will be cancelled. If a player breaks the block directly below their feet the block will reappear, but not before their feet have fallen slightly into the block below and they end up falling straight through.
I expected this to be blocked by passable, but passable does not seem to notice it happening at all.

If it can be reproduced please try to explain as detailed as possible (video, text or images are always welcome):
Speaking specifically when using HeavySpleef, though I assume this could apply to any plugin which cancels the break block event.

1. Build a lobby and area with single block high floors
2. Join an arena/game of spleef in the heavyspleef plugin. You are switched to survival and moved to the lobby/arena floor.
3. Before the game starts, stand in the centre of a block and break that block
4. The block will reappear but your feet will have fallen slightly into the top of the block and you will proceed to fall through it without being stopped or otherwise detected by passable.

All players are affected by this, and it happens consistently.
Tested only using a vanilla 1.9 client.

Have you made any (noteworthy) changes to the default configuration of NCP?
No.

Do you run any other special plugins that affect game mechanics next to NC+ (adding skills, machines, adding/changing blocks, other anti cheating)?
Only HeavySpleef is providing anti-block breaking.

 ---- Version information ----
 #### Server ####
 git-Spigot-f04e043-52c7fc0 ~MC: 1.9~
   detected: 1.9
 #### NoCheatPlus ####
 Plugin: 3.14.0-SNAPSHOT-sMD5NET-b955
 MCAccess: 1.9 / Spigot-CB-1.9_R1
 Features:
   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9
  checks: FastConsume | Gutenberg | AttackFrequency | FlyingFrequency | KeepAliveFrequency
   packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter
 Hooks: AllViolations~NCP~ 1.0 | BungeeNCPNotify 1.0
 #### Related Plugins ####
 ProtocolLib v3.7-SNAPSHOT-b236

**Comment by asofold** [comment]
As long as the block breaking events are cancelled server side, the blocks are still there on server side, so passable should prevent moving through them. To clarify, we are having ordinary full blocks in focus, like stone/wool/sand?

@MyPictures Could you reproduce this yet?

**Comment by zedwick** [comment]
The blocks tested are stained_glass and full snow blocks.

**Comment by asofold** [comment]
Passable should prevent moving through those (by specifications).

So question is, why they can fall through (bug, other plugin allows it, server teleports players into blocks?).

If you can reproduce it reliably, you could record a debug log:
1. 'ncp debug player PLAYER_NAME'
2. reproduce (preferable a couple of times)
3. 'ncp remove PLAYER_NAME'

Such could help identifying teleportation and what block shapes are reported server side.

**Comment by MyPictures** [comment]
I was not able to reproduce this with:
[10:08:16 INFO]: [NoCheatPlus] ºcNCP: ºfReloading configuration...
[10:08:16 INFO]: [NoCheatPlus] McAccess set to: 1.9 / Spigot-CB-1.9_R1
[10:08:16 INFO]: [NoCheatPlus] ºcNCP: ºfConfiguration reloaded.
[10:08:16 INFO]: [NoCheatPlus] Added block-info for Minecraft 1.5 blocks.
[10:08:16 INFO]: [NoCheatPlus] Added block-info for Minecraft 1.6.1 blocks.
[10:08:16 INFO]: [NoCheatPlus] Added block-info for Minecraft 1.7.2 blocks.
[10:08:16 INFO]: [NoCheatPlus] Added block-info for Minecraft 1.8 blocks.
[10:08:16 INFO]: [NoCheatPlus] Added block-info for Minecraft 1.9 blocks.
[10:08:16 INFO]: [NoCheatPlus] Removed hook: AllViolations(NCP) [1.0].
[10:08:16 INFO]: [NoCheatPlus] Added hook: AllViolations(NCP) [1.0].
[10:08:16 INFO]: [NoCheatPlus] Adding packet level hooks for ProtocolLib (MC 1.9.0)...
[10:08:16 INFO]: [NoCheatPlus] Disable SoundDistance due to incompatibilities.
[10:08:16 INFO]: [NoCheatPlus] Available (and activated) packet level hooks: DebugAdapter | UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter

Passable showed up as expected (Snow blocks): 
[16-03-28-1.zip](https://github.com/NoCheatPlus/Issues/files/191527/16-03-28-1.zip)

**Comment by zedwick** [comment]
Is there a permission node for /ncp debug? There isn't one listed on the permissions page.

**Comment by asofold** [comment]
The permission is: nocheatplus.command.debug

For removing data: nocheatplus.command.remove

These permissions allow to debug and remove data for all players.

Slightly more safe to use on oneself is nocheatplus.shortcut.tester, only allowing to remove your own data, but still allowing to view info about other players and start on the fly debug logging, so it's not really meant for random players (yet).

More admin like permissions would be nocheatplus.shortcut.admin, including the above permissions.

Note, that there is an extra 'player' with 'ncp debug player PLAYER_NAME'.

**Comment by asofold** [comment]
What happened with this issue? Is anyone able to reproduce this?

**Comment by asofold** [comment]
With build 1005 the implementation has changed. I suspect this isn't the cause, because the old implementation should have been good enough.

It could be vanilla or the 'anti bock breaking' of that spleef plugin. Best would be to check with debug logging on (and perhaps some latency, as it might not be reproducible with a perfect connection or a local test server).

**Comment by asofold** [comment]
Closing as 'superseded'  for both server and client versions. Please comment or open a new issue if the issue persists.
